scripts/rpm-upgrade.sh

```diff
@@ -132,34 +132,20 @@ else
 fi
 set -u
 
+# Any of the below steps could fail
+trap save_failure_logs ERR
+
+read -ra package_array <<< "$package_list"
+
 # Install previous release
-echo "$package_list" | xargs sudo "$pkg_cmd" "$pkg_cmd_options" install ||
+if sudo "$pkg_cmd" "$pkg_cmd_options" install "${package_array[@]}"; then
   bb_log_err "installation of a previous release failed, see the output above"
-#fi
+  save_failure_logs
```

Comment:
Here we are mixing approaches i.e. :
1. command failures handled by IF or || need an explicit `save_failure_logs` call
2. other command failures are handled by `trap save_failure_logs ERR`

Why don't we stick to one across all scripts? Makes it easier to read. And it's easy to document how we handle errors.

Reply:
done. Unsure if documentation is complete enough.